docs/Masonry.md

```diff
@@ -197,3 +197,105 @@ ReactDOM.render(
   document.getElementById('example')
 );
 ```
+
+### Masonry example with dynamically measured images
+
+Items in the list in basic example can only be measured once, which means client has to know all the
+sizes upfront before displaying. For cases when server cannot deliver such data we need to
+pre-measure images. In order to preserve correct optimized layout images can only be displayed one
+by one in the order they appear in the array, not in the order they are loaded.
+
+These specifics were taken into account in a small library
+[react-virtualized-image-measurer](https://github.com/kirill-konshin/react-virtualized-image-measurer)
+here is an example with dynamically measured images:
+
+```js
+import React from "react";
+import { render } from "react-dom";
+import {
+  CellMeasurer,
+  CellMeasurerCache,
+  createMasonryCellPositioner,
+  Masonry
+} from "react-virtualized";
+import ImageMeasurer from "react-virtualized-image-measurer";
+
+// Array of images with captions
+//const list = [{image: 'http://...', title: 'Foo'}];
+
+// We need to make sure images are loaded from scratch every time for this demo
+const noCacheList = list.map(item => ({
+  ...item,
+  image: item.image + "?noCache=" + Math.random()
+}));
+
+const columnWidth = 200;
+const defaultHeight = 250;
+const defaultWidth = columnWidth;
+
+const MasonryComponent = ({ itemsWithSizes }) => {
+  // Default sizes help Masonry decide how many images to batch-measure
+  const cache = new CellMeasurerCache({
```

Comment:
This should not be created within the `MasonryComponent` component because that would mean it gets recreated each time.